src/lib/requests/baseRequests.js

```diff
@@ -0,0 +1,280 @@
+            method: 'POST',
+            uri: buildUrl(this._pickPeerEndpoint(resourceType), url),
+            headers: {
+                ...this._buildHeaders('POST', resourceType, dest),
+                ...headers,
+            },
+            body: body,
+            qs: query,
+        };
+
+        if (responseType === ResponseType.Stream) {
+            reqOpts.responseType = request.responseType.Stream;
+        }
+
+        if ((responseType === ResponseType.Mojaloop) && this.jwsSign) {
+            this.jwsSigner.sign(reqOpts);
+        }
+
+        reqOpts.body = bodyStringifier(reqOpts.body);
+
+        this.logger.log(`Executing HTTP POST: ${util.inspect(reqOpts)}`);
+        return request({ ...reqOpts, agent: this.agent })
+            .then((res) => (responseType === ResponseType.Mojaloop) ? throwOrJson(res) : res)
+            .catch(e => {
+                this.logger.log('Error attempting POST. URL:', url, 'Opts:', reqOpts, 'Body:', body, 'Error:', e);
+                throw e;
+            });
+    }
+
+    /**
+     * @function _buildHeaders
+     * @description
+     *   Utility function for building outgoing request headers as required by the mojaloop api spec
+     * @param {'GET' | 'POST' | 'PUT'} method The HTTP Method
+     * @param {string} resourceType - The 'type' of resource, as defined in the Mojaloop specification
+     * @param {string | undefined} dest - The destination participant. Leave empty if participant is unknown (e.g. `GET /parties`)
+     *
+     * @returns {*} headers object for use in requests to mojaloop api endpoints
+     */
+    _buildHeaders(method, resourceType, dest) {
+        let headers = {
+            'content-type': `application/vnd.interoperability.${resourceType}+json;version=1.0`,
+            'date': new Date().toUTCString(),
+        };
+
+        if (this.dfspId) {
+            headers['fspiop-source'] = this.dfspId;
+        }
+
+        if(dest) {
+            headers['fspiop-destination'] = dest;
+        }
+
+        //Need to populate Bearer Token if we are in OAuth2.0 environment
+        if (this.wso2Auth) {
+            const token = this.wso2Auth.getToken();
+            if(token) {
+                headers['Authorization'] = `Bearer ${token}`;
+            }
+        }
+
+        // dont add accept header to PUT requests
+        if(method.toUpperCase() !== 'PUT') {
+            headers['accept'] = `application/vnd.interoperability.${resourceType}+json;version=1.0`;
+        }
+
+        return headers;
+    }
+
+    /**
+     * @function _pickPeerEndpoint
+     * @description Utility function for picking up the right endpoint based on the resourceType
+     * @param {string} resourceType - The 'type' of resource, as defined in the Mojaloop specification
+     * @returns {string} The endpoint fot the given `resourceType`
+     */
+    _pickPeerEndpoint(resourceType) {
+        // TODO: refactor to remove the need for all the damn question marks?
```

Comment:
maybe using a map (resurceType -> endpoint) object would simplify this flow? the map could be created by the constructor. There is a visible pattern that resourceType value can be used to create property name with `Endpoint` postfix (only exception are parties and participants cases)

Reply:
Yeah I was also think about this, but I was trying not to touch too much at one time.